manifest.js

```diff
@@ -0,0 +1,187 @@
+ * limitations under the License.
+ */
+
+// manifest.js
+
+let rulesetFilePaths = [];
+
+// Uploading and displaying manifest file
+const manifestFileInput = document.getElementById('manifestFileInput');
+manifestFileInput.addEventListener('change', (event) => {
+    const manifestFile = event.target.files[0];
+    if(manifestFile){
+        const reader = new FileReader();
+        reader.onload = function(e) {
+            try {
+                const manifestObject = JSON.parse(e.target.result);
+                let manifestSyntaxError = isValidManifest(manifestObject);
+                if(manifestSyntaxError === true){ // if manifestSyntaxError is true, i.e., there are no errors
+                    displayRulesetFilePaths(manifestObject);
+                } else {
+                    const fileInfoDiv = document.getElementById('manifestFileInfo');
+
+                    let output = "Issues found:\n";
+
+                    for (let i = 0; i < manifestSyntaxError['type'].length; i++) {
+                        if (manifestSyntaxError['type'][i] === 'missingFields') {
+                            let missingFields = manifestSyntaxError['missingFields'].join(', ');
+                            output += `- Missing fields: ${missingFields}\n`;
+                        }
+                        if (manifestSyntaxError['type'][i] === 'invalidValueTypes') {
+                            let invalidValueTypes = manifestSyntaxError['invalidValueTypes'].join(', ');
+                            output += `- Invalid value types for: ${invalidValueTypes}\n`;
+                        }
+                    }
+                    
+                    fileInfoDiv.innerText = output;
+                        
+                }
+            } catch(error){
+                console.log("Error parsing manifest JSON: ", error);
+                document.getElementById('manifestFileInfo').textContent = 'Error parsing manifest.json file';
+            }
+        }
+        reader.readAsText(manifestFile);
+    }
+});
+
+// Display the paths for the ruleset files, as defined in the manifest
+function displayRulesetFilePaths(manifest){
+    const fileInfo = document.getElementById('manifestFileInfo');
+    let output = 'Ruleset Files:\n';    
+    if(manifest.declarative_net_request.rule_resources){
+        manifest.declarative_net_request.rule_resources.forEach(ruleset => {
+            output += `- ${ruleset.id}: ${ruleset.path}, Enabled: ${ruleset.enabled}\n`;
+            rulesetFilePaths.push({rulesetFilePath: ruleset.path, rulesetId: ruleset.id});
+        });
+        // console.log(rulesetFilePaths); // correct
+        fileInfo.innerText = output;     
+    }       
+}
+
+// Checks syntax and validity of the manifest file
+function isValidManifest(manifest) {
+    let syntaxError = {};
+    syntaxError['type'] = [];
+
+    // Check for declarativeNetRequest permission
+    if(!manifest.permissions.includes('declarativeNetRequest')){
```

Comment:
We might also want to allow `declarativeNetRequestWithHostAccess` as an alternative, since you only need to specify one of the two. Let me know if you need any help understanding that.